include/pybind11/pytypes.h

```diff
@@ -890,6 +891,22 @@ class iterator : public object {
     object value = {};
 };
 
+class type : public handle {
+public:
+    // Explicit omitted here since PyTypeObject required (rather than just PyObject)
+    type(PyTypeObject* type_ptr) : handle((PyObject*) type_ptr) {}
```

Comment:
Should we not try doing this the standard way, with the `PYBIND11_OBJECT` or `PYBIND11_OBJECT_CVT` macros? Those will add `check`, etc. (It there's a reason to nót use it, some docs on why would be nice to have documented.)

Reply:
That would make this an `py::object` subclass instead of a `py::handle` sublcass. Is that what we want for type? (final question in the comment on the PR)

Reply:
Yeah, I think this should just follow suite with `py::dtype` and inherit either directly from `py::object` or `py::object_api`?

Reply:
FWIW `py::dtype` uses `PYBIND11_OBJECT_DEFAULT`, so I do think that this should use the type-checking constructor helper, as well as inherit from `py::object`.